spec.emu

```diff
@@ -763,6 +784,55 @@
           1. Let _rest_ be WithClauseToAttributes of |WithEntries|.
           1. Return the list-concatenation of « _entry_ » and _rest_.
         </emu-alg>
+
+        <emu-note>
+          <p><ins>This section is extended by <emu-xref href="#sec-deprecated-assert-keyword-for-import-attributes"></emu-xref>.</ins></p>
+        </emu-note>
+      </emu-clause>
+
+      <emu-clause id="sec-deprecated-assert-keyword-for-import-attributes" deprecated normative-optional>
+        <h1><ins>Deprecated `assert` keyword for Import Attributes</ins></h1>
+
+        <p>Import Attributes are denoted using the `with` keyword. However, there are existing implementations based on a previous version of the proposal using the `assert` keyword. Due to potential web compatibility risks, the `assert` keyword is still included in this specification, but its removal is being investigated.</p>
+
+        <p>Implementations that implement this clause must also implement support for the deprecated `assert` fallback for import attributes in import calls (step <emu-xref href="#step-evaluate-import-call-assert-fallback"></emu-xref> of <emu-xref href="#sec-evaluate-import-call"></emu-xref>).</p>
+
+        <p>The following augments the |ImportDeclaration| production in <emu-xref href="#sec-imports"></emu-xref>:</p>
+        <emu-grammar type="definition">
+          ImportDeclaration :
+            `import` ImportClause FromClause [no LineTerminator here] AssertClause `;`
+            `import` ModuleSpecifier [no LineTerminator here] AssertClause `;`
+
+          AssertClause :
+            `assert` `{` `}`
+            `assert` `{` WithEntries `,`? `}`
+        </emu-grammar>
+
+        <p>The static semantics of Early Errors in <emu-xref href="#sec-imports-static-semantics-early-errors"></emu-xref> are augmented with the following:</p>
+        <emu-grammar>
+          AssertClause : `assert` `{` WithEntries `,`? `}`
+        </emu-grammar>
+        <ul>
+          <li>It is a Syntax Error if WithClauseToAttributes of |AssertClause| has two entries _a_ and _b_ such that _a_.[[Key]] is _b_.[[Key]].</li>
+          <li>It is a Syntax Error if AllImportAttributesSupported(WithClauseToAttributes of |AssertClause|) is *false*.</li>
+        </ul>
+
+        <p>The static semantics of WithClauseToAttributes in <emu-xref href="#sec-with-clause-to-attributes"></emu-xref> are augmented with the following:</p>
```

Comment:
I don't think it's *unclear* that we're inserting these (as opposed to replacing the existing ones), but I think I'd prefer to be more explicit with wording like

```suggestion
        <p>The static semantics of WithClauseToAttributes in <emu-xref href="#sec-with-clause-to-attributes"></emu-xref> are augmented to include the following:</p>
```

Also, since we are disallowing these `AssertClause` nodes from being parsed in the first place using early errors, we don't technically need these definitions to be conditional. We could just put them in `WithClauseToAttributes`. @tc39/ecma262-editors thoughts?

Reply:
I think it's nice to keep the deprecated feature as much self contained as possible, rather than scattered across different parts of the spec.